### DISCUSSION TITLE
No IPv4 binding

### DISCUSSION THREAD

**Discussion by ChristophBSchell** [rustdesk/rustdesk#5695] [Q&A]
Hello,

I am using a rustdesk Pro server but nobody is able to vonnect to the server.
Ther server is build from the repros on a ubuntu 20.04 LTS, all updates are instaelled.
But the server is not binding any port on IPv4 only on IPv6

tcp6       0      0 :::21114                :::*                    LISTEN      700/hbbs
tcp6       0      0 :::21115                :::*                    LISTEN      700/hbbs
tcp6       0      0 :::21116                :::*                    LISTEN      700/hbbs
tcp6       0      0 :::21117                :::*                    LISTEN      699/hbbr
tcp6       0      0 :::21118                :::*                    LISTEN      700/hbbs
tcp6       0      0 :::21119                :::*                    LISTEN      699/hbbr
udp6       0      0 :::21116                :::*                                700/hbbs
udp6       0      0 :::34437                :::*                                700/hbbs

Allthough the server has an IPv4 address

2: enp7s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1450 qdisc fq_codel state UP group default qlen 1000
    link/ether 86:00:00:5a:fd:64 brd ff:ff:ff:ff:ff:ff
    inet 192.168.211.10/32 scope global dynamic enp7s0
       valid_lft 83969sec preferred_lft 83969sec
    inet6 fe80::8400:ff:fe5a:fd64/64 scope link
       valid_lft forever preferred_lft forever

Restarting the services and rebooting did not fix the problem.

How can I force the hbbr and hbbs to use IPv4?

**Comment by dinger1986** [comment]
Sorry what do you mean build with the repros?

**Reply by ChristophBSchell** [reply]
It is installed from thr repositories, it is not a docker installation.

**Reply by dinger1986** [reply]
Can you try installing with the install script?

**Reply by ChristophBSchell** [reply]
I used the script from the webpage

**Comment by SergeyMy** [comment]
`How can I force the hbbr and hbbs to use IPv4?`

Everything is fine
tcp6 0 0 ::: is a loopback address and your server is bound to all IPv4 and IPv6 addresses.
The problem is somewhere else, most likely in the firewall.

**Reply by ChristophBSchell** [reply]
It is not fine.
tcp6 is only for IPv6, but this server is not configured for IPv6

Service That are running on IPv4 and IPv6 do have 2 entries, for example sshd:
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      776/sshd: /usr/sbin
tcp6       0      0 :::22                   :::*                    LISTEN      776/sshd: /usr/sbin

tcp for IPv4 and tcp6 for IPv6.

Why are no tcp entries for hbbr and hbbs?

For testing the firewall on the server is disabled and is still not working.

**Reply by SergeyMy** [reply]
> but this server is not configured for IPv6

you're wrong - at least for the local network only
PS IPv6 non-disabled stack in modern systems

**Comment by ChristophBSchell** [comment]
Problem ist fixed after reinstall the whole server.
Thanks for your help

**Comment by ChristophBSchell** [comment]
done

**Comment by SirAry** [comment]
Did you succeeded after reinstall ubuntu? I have same problem after installing with any option (1. for detecting Wan IP or 2. Domain/dns) I get listening only on IPV6. I see that with netstat. and of course not connecting.

**Comment by ProxyAi** [comment]
Im getting the same issue from fresh install today on a freshly install debain 12 machine
